mula/scheduler/server/server.py

```diff
@@ -398,12 +402,12 @@ def list_tasks(
         except storage.filters.errors.FilterError as exc:
             raise fastapi.HTTPException(
                 status_code=fastapi.status.HTTP_400_BAD_REQUEST,
-                detail=str(exc),
+                detail=f"invalid filter(s) [exception: {exc}]",
             ) from exc
-        except ValueError as exc:
+        except storage.errors.StorageError as exc:
             raise fastapi.HTTPException(
-                status_code=fastapi.status.HTTP_400_BAD_REQUEST,
-                detail=str(exc),
+                status_code=fastapi.status.HTTP_500_INTERNAL_SERVER_ERROR,
+                detail=f"error occurred while accessing the database [exception: {exc}]",
```

Comment:
We could consider a generic `@app.exception_handler` as well, which would clean up the router code but forces you to combing thinking about Exception hierarchy with a hierarchy of exception_handlers.

Reply:
Yes I agree, let's pick it up in a new issue